### DISCUSSION TITLE
Ability to change nested blade component delimiter

### DISCUSSION THREAD

**Discussion by Wulfheart** [laravel/framework#33498] [General] [answered]
[Nested blade components](https://laravel.com/docs/7.x/blade#displaying-components) can be accessed like ``<x-inputs.button/>`` for a component located at ``App\View\Components\Inputs\Button.php``.
I'm using emmet to autogenerate my tags, so normally ``div.someClass`` becomes ``<div class="someClass"></div>``, when typing ``x-inputs.button`` then it becomes ``<x-inputs class="button"></x-inputs>``.
Maybe a config option to set a nesting delimiter.
If this discussion gets positive reactions I'd also submit a pull request.

**Comment by driesvints** [comment, marked as answer]
I don't think we'll be changing this any time soon sorry. I'm not familiar with emmet. What's keeping you exactly from using the current delimiter?

**Reply by Wulfheart** [reply]
emmet allows you to quick-generate html. Let's say we'd like to generate a div with a class "foo" we can type ``div.foo`` and it will be generated to ``<div class="foo"></div>``. There are also some more advanced possibilities like this example:
``ul>li*5>span.foo+customTag.bar*2`` becomes
````
<ul>
    <li>
        <span class="foo"></span>
        <customTag class="bar"></customTag>
        <customTag class="bar"></customTag>
    </li>
    <li>
        <span class="foo"></span>
        <customTag class="bar"></customTag>
        <customTag class="bar"></customTag>
    </li>
    <li>
        <span class="foo"></span>
        <customTag class="bar"></customTag>
        <customTag class="bar"></customTag>
    </li>
    <li>
        <span class="foo"></span>
        <customTag class="bar"></customTag>
        <customTag class="bar"></customTag>
    </li>
    <li>
        <span class="foo"></span>
        <customTag class="bar"></customTag>
        <customTag class="bar"></customTag>
    </li>
</ul>
````

In my opinion this is a huge timesaver as you can easily large structures. Furthermore is emmet supported out of the box by many code editors and IDEs. It can confirm that it works out of the box if activated in Visual Studio Code and PhpStorm without the need to install additional plugins.
I'm not asking for a complete change of the templating syntax. I'm asking for an ability to configure the framework. Something like 
````
'blade_nested_component_delimiter' => '/'
````
in e.g. ``config/app.php``. This way the convention stays the same but it is also configurable.

**Reply by driesvints** [reply]
Sorry but we're probably not gonna support that.

**Reply by Wulfheart** [reply]
Is it because of the overhead it generates?

**Reply by driesvints** [reply]
That and we're probably not going to take the effort just to support one library, sorry.

**Reply by Wulfheart** [reply]
It seems that emmet is less known than I thought. Thanks for taking the time.

**Reply by ryangjchandler** [reply]
I wouldn't say that emmet is less known, it's just people don't know it as emmet. This sort of autocompletion comes out of the boxes in a lot of editors, including VSCode and PhpStorm.

**Reply by Wulfheart** [reply]
And this autocompletion doesn’t allow escaping dots...😫
